### PR TITLE
[Key Vault] Address administration feedback

### LINKED ISSUE #19123: Resolve architects' review feedback for Key Vault
Feedback from our architects:

- [ ] Delete methods on the `KeyVaultAccessControlClient` should return None and ignore HTTP 404s.
- [ ] KeyVaultRestoreOperation and KeyVaultSelectiveKeyRestoreOperation can be removed
- [ ] `KeyVaultBackupClient.begin_restore` and `.begin_selective_key_restore` can be merged into a single method -- a `key_name` kwarg will determine the behavior
- [ ] KeyVaultBackupOperation should only have a `folder_url` property
- [ ] Restore operations should return `LROPoller[None]`, and backup operations should return `LROPoller[KeyVaultBackupOperation]`
- [ ] Instead of `get_backup_status` and `get_restore_status`, pollers should be rehydrate-able with a simplified continuation token (e.g. the value of the response's `azure-asyncoperation` header instead of a pickled PipelineResponse)
- [ ] `role_*` parameter names should be shortened as much as possible without ambiguity

Feel free to update this issue with idiomatic feedback.

### PR DESCRIPTION
**Pull request by mccoyp**
Summarizing changes:
- delete methods return None, do not raise on 404
- merge `begin_selective_key_restore` into `begin_restore`, former gains a `key_name` kwarg
- `begin_restore` poller result is None, i.e. no longer a model
- remove `get_backup_status`, `get_restore_status`
- remove models `KeyVaultRestoreOperation`, `KeyVaultSelectiveKeyRestoreOperation`
- `KeyVaultBackupOperation` reduced to one property, `folder_url`
- removed prefixes from unambiguous name parameters
    - e.g. `create_role_assignment(assignment_name=...)` -> `create_role_assignment(name=...)`

Closes #19123

### CODE REVIEW

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py [hunk 1]
```diff
@@ -40,45 +37,28 @@ def _from_generated(cls, permissions):
...
+    :ivar str type: type of the assignment
+    """
 
     def __init__(self, **kwargs):
         # type: (**Any) -> None
-        self._role_assignment_id = kwargs.get("role_assignment_id")
-        self._name = kwargs.get("name")
-        self._properties = kwargs.get("properties")
-        self._type = kwargs.get("assignment_type")
+        self.name = kwargs.get("name")
```
Comment (mccoyp) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py:50]: Is there a particular reason to move away from properties?

Comment (chlowell) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py:50]: Architect opinion: https://github.com/Azure/azure-sdk/blob/master/docs/python/_includes/example_client.py#L26

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py [hunk 2]
```diff
@@ -29,100 +30,103 @@ class KeyVaultAccessControlClient(KeyVaultClientBase):
...
 
-        :param role_scope: scope of the role definition. :class:`KeyVaultRoleScope` defines common broad scopes.
+        To update a role definition, provide the ``definition_name`` of the existing definition.
+
+        :param scope: scope of the role definition. :class:`KeyVaultRoleScope` defines common broad scopes.
             Specify a narrower scope as a string. Managed HSM only supports '/', or KeyVaultRoleScope.GLOBAL.
-        :type role_scope: str or KeyVaultRoleScope
-        :param role_definition_name: the unique role definition name. Unless a UUID is provided, a new role definition
+        :type scope: str or KeyVaultRoleScope
+        :keyword definition_name: the unique role definition name. Unless a UUID is provided, a new role definition
```
[the remark this answers is not in the document]
Comment (chlowell) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:126]: Seems reasonable to me, let's give it a try.

File: sdk/keyvault/azure-keyvault-administration/CHANGELOG.md [hunk 3]
```diff
@@ -32,6 +46,11 @@
   - `denied_actions` -> `not_actions`
   - `allowed_data_actions` -> `data_actions`
   - `denied_data_actions` -> `denied_data_actions`
+- Renamed argument `role_assignment_name` to `name` in
+  `KeyVaultAccessControlClient.create_role_assignment`, `.delete_role_assignment`,
+  and `.get_role_assignment`
+- Renamed argument `role_definition_name` to `name` in
+  `KeyVaultAccessControlClient.delete_role_definition` and `.get_role_definition`
```
Comment (mccoyp) [inline sdk/keyvault/azure-keyvault-administration/CHANGELOG.md:54]: We'll want to mention other argument renames too, like `role_scope` -> `scope` in multiple methods

**Review by mccoyp — COMMENTED** [review]
Looks good! I think we need to also override a frozen requirement to get CI passing: https://dev.azure.com/azure-sdk/public/_build/results?buildId=942747&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=b64fce29-1009-5e76-50fb-7b596c2db4c3&l=2026

**Review by chlowell — APPROVED** [review]

**Pull request merged**